File: lab/javascript/matrizes/busca_matriz.js
function busca_item_matriz(matriz, valor){
    // Se a matriz for vazia, ele retorna false
    if (matriz.length === 0){
        return false;
    }

    for (let linha=0; linha < matriz.length; linha++){
        for (let coluna=0; coluna < matriz.length; coluna++){
            if (matriz[linha][coluna] == valor){
                return true;
            }
        }
    }

    return false;
}


const testes = [
    {"matriz": [[1, 2], [3, 4]], "valor": 3},
    {"matriz": [[1, 2], [3, 4]], "valor": 7}
]

for (const teste of testes){
    if (busca_item_matriz(teste.matriz, teste.valor)){
        console.log("Encontrado");
    } else {
        console.log("Não encontrado");
    }
}

File: lab/python/matrizes/busca_matriz.py
def busca_item_matriz(matriz, valor):
    # Se a matriz for vazia, ele retorna false
    if not matriz:
        return False

    for linha in range(len(matriz)):
        for coluna in range(len(matriz)):
            if matriz[linha][coluna] == valor:
                return True

    return False


testes = [
    ([[1, 2], [3, 4]], 3),
    ([[1, 2], [3, 4]], 7)
]

for teste in testes:
    matriz, busca = teste
    if busca_item_matriz(matriz, busca):
        print("Encontrado")
    else:
        print("Não encontrado")
